pupil_src/shared_modules/gl_utils/draw.py

```diff
@@ -0,0 +1,50 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2021 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+import functools
+import typing as T
+
+import numpy as np
+import OpenGL.GL as gl
+from pyglui.cygl.utils import RGBA
+
+
+def draw_circle_filled(
+    screen_point: T.Tuple[float, float], size: float, color: RGBA, num_points: int = 50
+):
+    points = _circle_points_offset(
+        screen_point, radius=size, num_points=num_points, flat=False
+    )
+    gl.glColor4f(color.r, color.g, color.b, color.a)
+    gl.glEnableClientState(gl.GL_VERTEX_ARRAY)
+    gl.glVertexPointer(2, gl.GL_DOUBLE, 0, points)
+    gl.glDrawArrays(gl.GL_POLYGON, 0, points.shape[0])
+
+
+@functools.lru_cache(4)  # 4 circles needed to draw calibration marker
+def _circle_points_offset(
```

Comment:
We should test how often we get cache-misses here for the calibration markers. It is possible that with a previous calibration, the gaze visualization clears the cache constantly. Maybe it would make sense to keep this function uncached and wrap it in the calibration marker vis:
```py
self._circle_points_offset = functools.lru_cache(4)(_circle_points_offset)
```